### ISSUE TITLE
Allow relative path server URLs

### ISSUE THREAD

**Issue by mgagliardo91** [danielgtaylor/restish#5]
We are using swagger to comply with the OpenAPI 3.0 spec and currently have outlined a single `server`:

```yaml
servers:
  - url: /v1
    name: API Version 1
```

We use this configuration to enable us to set the paths on each endpoint to be version-agnostic, and the swagger interface (according to the openapi spec) will use the relative path `/v1` as a prefix for each of our defined paths against the current hostname being accessed.

We are hoping to gain the same functionality out of the restish configuration. When we go to add an `api` to restish, we are explicitly indicating the protocol/hostname of the API we are trying to access. Even though we do have the `/v1` server setup, the following code indicates that it will be ignored since it is not prefixed with an absolute `${scheme}://${host}`, which results in all of our paths being requested against the root of our api server instead of the path `/v1`.

```go
	// See if this server has any base path prefix we need to account for.
	// TODO: handle variables in the server path?
	basePath := ""
	prefix := location.Scheme + "://" + location.Host
	for _, s := range swagger.Servers {
		if strings.HasPrefix(s.URL, prefix) {
			base, err := url.Parse(s.URL)
			if err != nil {
				return cli.API{}, err
			}
			basePath = base.Path
		}
	}
```

**Proposed Change:**

Modify the above code to respect relative server URLs. If the `serverUrl` is an absolute path, continue with the `HasPrefix` check, if not, set the basePath to the relative route `/v1`.

Thanks for your help, this should be a quick fix and will help get us up and running!

**Comment by danielgtaylor** [comment]
This is an interesting use-case I hadn't considered.

**Comment by danielgtaylor** [comment]
Merged in 5a6424d518891f97b0979967ae975e83f3392d0a and released in 0.4.3.

**Comment by asaf** [comment]
Hey,

@danielgtaylor this is still not working when configuring an API via `~/.restish/apis.json`

example:
```json
{
  "myapi": {
    "base": "http://domain.com:8080/api/v1",
    "spec_files": ["/Users/home/openapi.yml"]
  }
}
```

any call via restish is sent to `http://domain.com:8080`.

removing the base causes nil pointer exception:

```go
...
github.com/danielgtaylor/restish/cli.Run.func1()
	/home/runner/work/restish/restish/cli/cli.go:461 +0x94
panic(0x1869040, 0xc00040bd70)
	/opt/hostedtoolcache/go/1.14.9/x64/src/runtime/panic.go:969 +0x166
github.com/danielgtaylor/restish/cli.MakeRequestAndFormat(0xc00009a900)
	/home/runner/work/restish/restish/cli/request.go:344 +0x100
github.com/danielgtaylor/restish/cli.Operation.command.func1(0xc0002042c0, 0xc000427740, 0x0, 0x1)
```


Thanks.
